src/Imgix/UrlBuilder.cs

```diff
@@ -56,40 +58,217 @@ public String BuildUrl(String path)
             return BuildUrl(path, new Dictionary<string, string>());
         }
 
+        /// <summary>
+        /// Create a resolvable imgix url.
+        /// 
+        /// </summary>
+        /// <param name="path">path to the image, i.e. "image/file.png"</param>
+        /// <param name="parameters">dictionary of query parameters</param>
+        /// <returns>resolvable URL</returns>
+
+        /*
+        * Encodes the `path` and `paramater` arguments to the UTF8 
+        * scheme and returns a formatted string that conforms to the 
+        * "{scheme}://{domain}/{path}{params}" format.
+        */
         public String BuildUrl(String path, Dictionary<String, String> parameters)
```

Comment:
I had a bit of difficulty parsing out the steps of this method despite knowing what it's supposed to do, specifically why `qs` was used over `localParams` at times (and vice versa). I think stylistically we can make things clearer for other contributors by either 1) reworking the variable names to be a bit more descriptive (I'll mull this one over) and/or 2) adding a comment describing what the purpose of the `if` block on [L82](https://github.com/imgix/imgix-csharp/pull/40/commits/8068ee09051f2eb0558d3db453630f3b267b140c#diff-7ae8782884590f68d4dc070e0791a2d2d4c30a536f3ed7c2bc46728467b729e7R82) is for.

Reply:
This makes sense. I think the best thing to do here is to break this up similarly to [js-core](https://github.com/imgix/js-core/blob/a213da0f94c261129be4408e775c54885d79c389/src/index.js#L47-L60), where sanitizePath, buildParams, and signParams are all extracted out into methods. Makes reading this a lot easier and also allows us remove references to `qs` or `queryString` and just say `finalParams`.